packages/muze-axis/src/cartesian-axis/props.js

```diff
@@ -7,11 +8,28 @@ export const PROPS = {
     config: {
         sanitization: (context, value) => {
             const oldConfig = Object.assign({}, context._config || {});
-            value = mergeRecursive(oldConfig, value);
+            const mockedOldConfig = mergeRecursive({}, oldConfig);
+            value = mergeRecursive(mockedOldConfig, value);
+
             value.axisNamePadding = Math.max(value.axisNamePadding, 0);
-            if (value.orientation !== oldConfig.orientation) {
-                context.axis(context.createAxis(value));
+            const shouldAxesScaleUpdate = hasAxesConfigChanged(
```

Comment:
This entire function can be broken down into logical components, each should act as a pure function.